operator/controllers/keptnworkloadinstance/controller.go

```diff
@@ -128,6 +128,11 @@ func (r *KeptnWorkloadInstanceReconciler) Reconcile(ctx context.Context, req ctr
 	appPreEvalStatus := appVersion.Status.PreDeploymentEvaluationStatus
 	if !appPreEvalStatus.IsSucceeded() {
 		if appPreEvalStatus.IsFailed() {
+			//cancel everything, as app pre-eval tasks have failed
+			if err := r.cancelWorkloadInstanceActions(ctx, workloadInstance, phase); err != nil {
```

Comment:
I'm not convinced of this, as the application pre-evaluation should only have the impact that the workload does not get scheduled. How do you deal with this when the workloads get resumed after the app pre-deployment evaluation/task passes? 

I would introduce a state "AppPreDeploymentTaskFailed/AppPreDeploymentEvaluationFailed" and leave the state untouched at the moment as this is not canceled.

Reply:
is not modifiying the workloads at all an option for you? In case AppPreEvaluationChecks fail, we will simple leave the workloads as they are... Just cancel the phases in the AppVersion

Reply:
if not, I would say let's discuss this in the team to have a clear vision how it should work